src/Controller.ts

```diff
@@ -53,12 +53,25 @@ export const Controller = {
 		return engines;
 	},
 
-	getFilteredEngines: async (filteredNames: string[]): Promise<RuleEngine[]> => {
+	getUserFacingEngines: async (): Promise<RuleEngine[]> => {
```

Comment:
Can we brainstorm some names for this during standup?

Reply:
I still haven't fixed the part where we plan to recognize `--engine pmd` to be `pmd-custom` if `--pmdconfig` is provided. This is mainly stemming from my indecision on how to handle the same for `eslint-custom`:

Should we handle any eslint-based engine in filter as `eslint-custom` because `--eslintconfig` is provided? I feel we shouldn't because we appear to be setting an unrealistic expectation from users that their dependencies will be taken care of when we clearly won't. It also feels weird to allow only `--engine eslint` to be treated as `eslint-custom` when `--eslintconfig` is provided.

Please let me know your thoughts.

Reply:
Update - this has been replaced by new filtering logic from the implementation of `isEngineRequested()` method by each engine.